crates/bitwarden-core/src/error.rs

```diff
@@ -4,6 +4,10 @@ use thiserror::Error;
 #[error("The response received was missing a required field: {0}")]
 pub struct MissingFieldError(pub &'static str);
 
+#[derive(Debug, Error)]
+#[error("The client vault is locked and needs to be unlocked before use")]
+pub struct VaultLocked();
```

Comment:
Nit: You can define the struct without parentheses and that way you don't need to use them every time you refer to this error type:
```suggestion
pub struct VaultLocked;
```